plugins/woocommerce/src/Blocks/BlockTypes/CollectionFilters.php

```diff
@@ -72,6 +72,34 @@ protected function enqueue_data( array $attributes = [] ) {
 		}
 	}
 
+	/**
+	 * Render the block.
+	 *
+	 * @param array    $attributes Block attributes.
+	 * @param string   $content    Block content.
+	 * @param WP_Block $block      Block instance.
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		$attributes_data = array(
+			'data-wc-interactive' => wp_json_encode( array( 'namespace' => 'woocommerce/collection-filters' ) ),
+			'class'               => 'wc-block-collection-filters',
+		);
+
+		if ( ! isset( $block->context['queryId'] ) ) {
+			$attributes_data['data-wc-navigation-id'] = sprintf(
+				'wc-collection-filters-%s',
+				md5( wp_json_encode( $block->parsed_block['innerBlocks'] ) )
```

Comment:
I think we need it to be persistent between the page loads for DOM diffing to work, right?

Reply:
As we discussed yeah unique_id may not be good here because if someone else calls it unexpectedly we'll get reorderd ids. its not a stable ID and we need that for the navigation ids